### DISCUSSION TITLE
Sync attached files

### DISCUSSION THREAD

**Discussion by workgroupengineering** [Mimetis/Dotmim.Sync#836] [Q&A] [answered]
Hi,

After a long time I am starting to play with this framework again.

Considering the following model in SQLite:

```csharp
    public class Document
    {
        [AutoIncrement]
        public Guid Id { get; set; }
        [NotNull]
        public string Description { get; set; }
        [NotNull]
        public string Path { get; set; } // It is relative path of document. The path is sub folder of app.
        [NotNull]
        public DateTime Expiration { get; set; }
        public DateTime? LastPrint { get; set; }
    }
```
Is there a way to tell Dotmim.Sync to sync attached accessory files (Document Path)?

**Comment by Mimetis** [comment, marked as answer]
ah ah, no sorry, DMS is a sync framework for relational databases, not files ;)

**Reply by workgroupengineering** [reply]
Hi, thanks for the reply.

The intent is not to make a file sync. When the record is modified, it is assumed that the file has been modified. Path is a workaround of using Sql Server's [FILESTREAM](https://learn.microsoft.com/en-us/sql/relational-databases/blob/filestream-sql-server?view=sql-server-ver16)s. I don't want to use a blob field because the files can be very large.

Is there a way to intercept Sync's CRUD operation and add other data?

**Reply by Mimetis** [reply]
You can look the https://dotmimsync.readthedocs.io/Interceptors.html#applying-changes section, you have several interceptors that can help you here I guess (like https://dotmimsync.readthedocs.io/Interceptors.html#onrowschangesapplying maybe)

**Reply by Mimetis** [reply]
Filestream is not supported, as it's a special feature only supported by SQL Server
